orttraining/orttraining/python/training/orttrainer_options.py

```diff
@@ -188,6 +188,15 @@ class ORTTrainerOptions(object):
                             'type': 'integer',
                             'min': 0,
                             'default': 0
+                        },
+                        'propagate_cast_ops_level': {
+                            'type': 'integer',
+                            'default': -1
```

Comment:
Why `-1`? Is it possible to use a enumeration or something more intuitive? Public facing APIs must be as intuitive as possible for users. Hiding internal low level constants from users ease adoption

Reply:
Optimization level numeric like -O0, -O1,etc. We are reserving 0 for the used defined list of allowed opcodes for cast propagation. Hence I had to choose something different,  -1, from 0, 1, 2,..., to to not run cast propagate transformation.

Reply:
This is the default value. The use doesn't need to set to -1.